packages/blind_pool/examples/future_polling_with_pooled_mut.rs

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+}
+
+/// A dummy waker for this example.
+struct DummyWaker;
+
+impl std::task::Wake for DummyWaker {
+    fn wake(self: std::sync::Arc<Self>) {
+        // In a real async runtime, this would wake up the executor
+    }
+}
+
+fn main() {
+    println!("Demonstrating PooledMut with Future polling");
+    println!("==========================================\n");
+
+    // Create a blind pool
+    let pool = BlindPool::new();
+
+    // Insert a future using insert_mut - this gives us exclusive mutable access
+    let mut future_handle = pool.insert_mut(CountdownFuture::new(3));
+
+    // Create a dummy waker and context for polling
+    let waker = Waker::from(std::sync::Arc::new(DummyWaker));
+    let mut context = Context::from_waker(&waker);
+
+    println!("Starting to poll the future...\n");
+
+    // Poll the future until completion
+    // We can do this because PooledMut implements DerefMut, giving us &mut access
+    loop {
+        // Pin the mutable reference to poll the future
+        let pinned_future = Pin::new(&mut *future_handle);
+
+        match pinned_future.poll(&mut context) {
+            Poll::Ready(result) => {
+                println!("Future completed with result: {result}");
+                break;
+            }
+            Poll::Pending => {
+                println!("Future returned Pending, continuing...\n");
+            }
+        }
+    }
+
+    println!("\nExample completed!");
+    println!("Pool length before cleanup: {}", pool.len());
+
+    // When we drop the handle, the future is automatically cleaned up
+    drop(future_handle);
+    println!("Pool length after cleanup: {}", pool.len());
+
+    // Demonstrate that we can also use insert_with_mut for in-place construction
+    println!("\nDemonstrating insert_with_mut...");
+
+    // SAFETY: We properly initialize the future in the closure.
+    let mut future_handle2 = unsafe {
+        pool.insert_with_mut(|uninit| {
+            uninit.write(CountdownFuture::new(2));
+        })
+    };
+
+    // Poll this one too
+    let waker2 = Waker::from(std::sync::Arc::new(DummyWaker));
+    let mut context2 = Context::from_waker(&waker2);
+
+    loop {
+        let pinned_future = Pin::new(&mut *future_handle2);
+        match pinned_future.poll(&mut context2) {
+            Poll::Ready(result) => {
+                println!("Second future completed: {result}");
+                break;
+            }
+            Poll::Pending => {
+                println!("Second future pending...");
+            }
+        }
+    }
+
+    // Demonstrate type casting to Future trait object
+    println!("\nDemonstrating casting to trait object...");
+
+    let mut future_handle3 = pool.insert_mut(CountdownFuture::new(1));
+
+    // We can cast to a trait object for dynamic dispatch
+    // Note: We use Pin::new_unchecked here because we know the future won't move
+    let future_ref: &mut dyn Future<Output = String> = &mut *future_handle3;
```

Comment:
It looks like you did not update the macro in `cast.rs` to also support this type. Conversion to trait objects is expected to occur via the `define_pooled_dyn_cast!` macro - ensure that it also works in this case.